lib/tests/streamlit/runtime/uploaded_file_manager_test.py

```diff
@@ -164,3 +165,118 @@ def test_cache_stats_provider(self):
             ),
         ]
         self.assertEqual(expected, self.mgr.get_stats())
+
+
+class UploadedFileManagerThreadingTest(unittest.TestCase):
+    # The number of threads to run our tests on
+    NUM_THREADS = 50
+
+    def setUp(self) -> None:
+        self.mgr = UploadedFileManager()
+
+    def test_add_file(self):
+        """`add_file` is thread-safe."""
+        # Call `add_file` from a bunch of threads
+        added_files = []
+
+        def add_file(index: int) -> None:
+            file = UploadedFileRec(
+                id=0, name=f"file_{index}", type="type", data=bytes(f"{index}", "utf-8")
+            )
+            added_files.append(self.mgr.add_file("session", f"widget_{index}", file))
+
+        call_on_threads(add_file, num_threads=self.NUM_THREADS)
+
+        # Ensure all our files are present
+        for ii in range(self.NUM_THREADS):
```

Comment:
why do you use `ii` instead of just `i` here, is it made on purpose, maybe I missing something

Reply:
Just personal habit - a holdover from a previous job where this was the standard. (I prefer `ii` because it's easier to search for, but I don't care strongly.)